tests/core/test_db.py

```diff
@@ -21,47 +35,76 @@ def test_check_database():
 
 
 @pytest.mark.parametrize(
-    "patch_settings",
-    [{"ENVIRONMENT": "demo", "AUTO_CREATE_SCHEMA": True}],
+    ("patch_settings", "caplog"),
+    [
+        ({"ENVIRONMENT": "demo", "AUTO_CREATE_SCHEMA": True}, None),
+        ({"ENVIRONMENT": "demo", "AUTO_CREATE_SCHEMA": False}, None),
+        ({"ENVIRONMENT": "production"}, None),
+    ],
     indirect=True,
 )
-def test_init_database_none(patch_settings: Settings):
-    org_exec = Session.exec
-    Session.exec = MagicMock()
-    Session.exec.return_value.first.return_value = None
-
+def test_init_database(patch_settings: Settings, caplog: pytest.LogCaptureFixture):
     init_db()
-
-    expected = [
-        (select(User).where(User.name == "Robbert"),),
-        (select(Status).where(Status.name == "Todo"),),
-        (select(Task).where(Task.title == "First task"),),
-    ]
-
-    for i, call_args in enumerate(Session.exec.call_args_list):
-        assert str(expected[i][0]) == str(call_args.args[0])
-
-    Session.exec = org_exec
-
-
-@pytest.mark.parametrize(
-    "patch_settings",
-    [{"ENVIRONMENT": "demo", "AUTO_CREATE_SCHEMA": True}],
-    indirect=True,
-)
-def test_init_database(patch_settings: Settings):
-    org_exec = Session.exec
-    Session.exec = MagicMock()
-
-    init_db()
-
-    expected = [
-        (select(User).where(User.name == "Robbert"),),
-        (select(Status).where(Status.name == "Todo"),),
-        (select(Task).where(Task.title == "First task"),),
-    ]
-
-    for i, call_args in enumerate(Session.exec.call_args_list):
-        assert str(expected[i][0]) == str(call_args.args[0])
-
-    Session.exec = org_exec
+    if patch_settings.ENVIRONMENT == "demo":
+        assert [record.message for record in caplog.records if "Creating demo data" in record.message]
+    if patch_settings.AUTO_CREATE_SCHEMA is True:
+        assert [record.message for record in caplog.records if "Creating database schema" in record.message]
+    assert [record.message for record in caplog.records if "Finished initializing database" in record.message]
+
+
+def test_remove_old_demo_objects(db: DatabaseTestUtils, monkeypatch: pytest.MonkeyPatch):
+    user = User(name="Robbert", avatar=None)
+    status = Status(name="Todo", sort_order=1)
+    task = Task(title="First task", description="This is the first task", sort_order=1, status_id=status.id)
+
+    db.given([user, status, task])
+    session_delete_counter = count_calls(monkeypatch, db.get_session(), db.get_session().delete)
```

Comment:
Why do we not use magicmock for the delete function here if you just want to know the amount of calls? 

https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock.call_count

Reply:
I want to know the logic works, so no exception is thrown f.e. and to make sure t is called as often as expected I use the counter.